dmarc.py

```diff
@@ -205,10 +206,25 @@ def parse(self):
         for f in glob.glob('./reports/*'):
             fp = None
             mime = filetype.guess_type(f)[0]
+            mime_tmp = ""
+
+            extension = f[f.rfind(".",0,len(f))+1:len(f)]
+           
             if mime is None:
                 print('ERROR: ', f)
                 continue
-            if mime in MIME_GZIP:
+
+            if mime in MIME_XML:
+                if extension == "gz":
+                    mime_tmp = mime
+                    mime = MIME_GZIP[0]
+                elif extension == "zip":
+                    mime_tmp = mime
+                    mime = MIME_ZIP[0]
```

Comment:
I add this because when I receive a report sent by Yahoo, the mimetype of the 'gz' file is 'application/xml' but the file is a real archive.